### DISCUSSION TITLE
Release Updates: 0.1.1 - 2024-03-17

### DISCUSSION THREAD

**Discussion by ykeremy** [Skyvern-AI/skyvern#97] [Announcements]
# _You Asked, We Listened: Run Skyvern with any LLM_

### TL;DR
- Introduced a new LiteLLM-powered module configurable with 100+ LLMs.
- Improved setup script for easier installation and configuration.
- Updated environment variables for better LLM provider setup.
- Fixed navigation delay issue and environment activation in setup.

### This Release's Soundtrack
[Setting the Tempo with Skyvern: Harder, Better, Faster, Stronger by Daft Punk](https://www.google.com/search?q=youtube+harder+better+faster+stronger&oq=youtube+harder+b&sourceid=chrome&ie=UTF-8#fpstate=ive&vld=cid:3e1053b7,vid:gAjR4_CbPpQ,st:0)

### Wanna contribute?
If you're looking to start contributing to Skyvern, look no further. Here are some issues we think are great:
- [Budget management for LLM calls](https://github.com/Skyvern-AI/skyvern/issues/70)
- [Dockerize Skyvern](https://github.com/Skyvern-AI/skyvern/issues/36)
- [Quickstart instructions for Windows and Linux platforms](https://github.com/Skyvern-AI/skyvern/issues/68)
- [Implementing an uninstall script](https://github.com/Skyvern-AI/skyvern/issues/80)

## Added
1. **A brand new LLM module:**

    - We got rid of our OpenAI client wrapper and built a new LLM module to provide an easy, extensible and configurable way to call LLM APIs.
    - This module is powered by [LiteLLM](https://github.com/BerriAI/litellm) and allows us call 100+ LLMs using the OpenAI format. We just loved how easy it is to use and all the features it comes bundled with. We've already started using its cost calculator feature. Moreover, we noticed that it also provides a Budget Manager. 
    - The biggest benefit of this new module is that it allows everyone to have a chance at running Skyvern with whichever LLM they have access to. We're looking forward to contributions to add more LLM configurations from the community. 
        - We already added support for new LLMs!
            - OpenAI GPT4 Vision
            - Azure OpenAI GPT4 Vision
            - Anthropic Claude 3 Opus
            - AND OpenAI GPT4 Turbo. Yes, now you have the option to observe Skyvern try to do its best without actually seeing the webpages! If you have a chance to try it out, [join our Discord channel](https://discord.com/invite/fG2XXEuQX3) and let us all know how it goes.
        - **_DISCLAIMER:_** Keep in mind that we didn't exhaustively test LLMs other than OpenAI GPT4 Vision and **_they may just not work as well as GPT4V does_**. 

1. **A live star history to our README**

    - We are grateful for the warm welcome from the community. As the founding team of Skyvern, we are avid believers that we'll go further if we go together. With this mindset, we recently open sourced a huge chunk of our code and [launched on Hackernews only 2 days ago](https://news.ycombinator.com/item?id=39706004). 

<p align="center">
  <a href=https://star-history.com/#Skyvern-AI/skyvern&Date> <img width="420" height="300" src="https://github.com/Skyvern-AI/skyvern/assets/26116031/c52ee8e2-8536-4f79-b670-8abfa5f78ef3"></a>
</p>

## Changed
- Setting up Skyvern is easier with a setup script that's better, faster, stronger 🙃!
    - It's running the PostgreSQL in a Docker container (if you don't already have one running on your local machine)! We noticed that setups don't always go as expected. We decided to level the playground instead of telling you guys it worked on our machines!
    - Now featuring a step by step LLM provider setup for your local environment. Here's an example execution:
<p align="center">
  <img width="520" height="400" src="https://github.com/Skyvern-AI/skyvern/assets/26116031/8d5cb71a-96a9-44d3-b588-0bb658b8fe72">
</p>

- Updated the `.env.example` with explanations for the new LLM provider environment variables.

## Fixed
- Added a small (3 seconds) delay before we navigate to a URL to make sure the browser is loaded correctly.
- Fixed an issue where the poetry environment wasn't always activated successfully. This caused some commands to fail and ultimately fail the setup script.

<hr /><em>This discussion was created from the release <a href='https://github.com/Skyvern-AI/skyvern/releases/tag/0.1.1'>0.1.1 - 2024-03-17</a>.</em>
